### DISCUSSION TITLE
Should result data be stored on Speckle and if, then how should it be structured?

### DISCUSSION THREAD

**Discussion by pfmephisto** [linkarkitektur/SpeckleLCA#26] [General]
I think it would be a good Idea to store geometry and results in the same place. This would make it easier for other apps/tools to work with the LCA results that we generate and it also would solve the issue of GDPR and data access.

The question I have is on how to structure the result data on Speckle. 
I assume that for any given commit we might end up with multiple results based on the materiale configuration in the materiale mapper.
The options I see are to either commit the results to the same branch as the geometry or to make a new dedicated results branch. 

But the main challenge in either approach is how to link geometry and result data. 
If commits are amendable it might be easy to just add pointers to the new result commits and in the results we can embed a reference to the geometry.

[_I have opened a question on the technicalities on the Speckle Forum_](https://speckle.community/t/best-approach-to-store-commit-related-analysis-results/5484?u=mephisto)

**Comment by rhughes42** [comment]
Have you seen this?

@pfmephisto 

Now, regarding your analysis results, you can commit them to a separate branch for better organization. The objects you send will sit alongside the source objects in the database, but the commits can be organized in a way that makes sense for you and your team.

When you analyse an object, the object remains unchanged, so the analysis results can be tightly coupled to that object’s speckle_id. You can maintain this coupling by caching or persisting it yourself, referring to the source object in the committed results objects or committing objects that refer to both the source and the results.

I believe this is the approach that is applied in act | a carbon tool: [https://act.speckle.arup.com/ 2](https://act.speckle.arup.com/)

The repo is open source, it might help you.
